apiserver/rest_test.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package apiserver_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"mime"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"runtime"
+
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+	"gopkg.in/juju/charm.v6-unstable"
+
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/testcharms"
+	jujuFactory "github.com/juju/juju/testing/factory"
+)
+
+// restCommonSuite wraps authHTTPSuite and adds
+// some helper methods suitable for working with the
+// charms endpoint.
+type restCommonSuite struct {
+	authHTTPSuite
+}
+
+func (s *restCommonSuite) restURL(c *gc.C, path string) *url.URL {
+	uri := s.baseURL(c)
+	uri.Path = fmt.Sprintf("/model/%s/rest/1.0/%s", s.modelUUID, path)
+	return uri
+}
+
+func (s *restCommonSuite) restURI(c *gc.C, path string) string {
+	return s.restURL(c, path).String()
+}
+
+func (s *restCommonSuite) assertGetFileResponse(c *gc.C, resp *http.Response, expBody, expContentType string) {
+	body := assertResponse(c, resp, http.StatusOK, expContentType)
+	c.Check(string(body), gc.Equals, expBody)
+}
+
+func (s *restCommonSuite) assertErrorResponse(c *gc.C, resp *http.Response, expCode int, expError string) {
+	charmResponse := s.assertResponse(c, resp, expCode)
+	c.Check(charmResponse.Error, gc.Matches, expError)
+}
+
+func (s *restCommonSuite) assertResponse(c *gc.C, resp *http.Response, expStatus int) params.CharmsResponse {
+	body := assertResponse(c, resp, expStatus, params.ContentTypeJSON)
+	var charmResponse params.CharmsResponse
+	err := json.Unmarshal(body, &charmResponse)
+	c.Assert(err, jc.ErrorIsNil, gc.Commentf("body: %s", body))
+	return charmResponse
+}
+
+type restSuite struct {
+	restCommonSuite
+}
+
+var _ = gc.Suite(&restSuite{})
+
+func (s *restSuite) SetUpSuite(c *gc.C) {
+	// TODO(bogdanteleaga): Fix this on windows
```

Comment:
Please write a better skip message. This bug is fix released.  Also, you could just say skipping becase apiserver will always be linux.  Perhaps better to test for linux rather than not windows.